### ISSUE TITLE
Could I use my own model to train?

### ISSUE THREAD

**Issue by Lavi11C** [shijxcs/LIFT#11]
In the beginning, I used vit-l14 and Imagenet to pretrain on other code. And now I wanna use your code to lightweight fine-tune the pretrain model.
But I just found I can test an existing model in the opt. I wonder whether I can train my own model with some setting or I have to rewrite some of your code.

Thanks firstly for reply.

**Comment by shijxcs** [comment]
It depends on whether your model can be reconstructed by `class Peft_ViT` in [`models/peft_vit.py`](https://github.com/shijxcs/LIFT/blob/main/models/peft_vit.py). We have implemented ViT models from two sources: `clip.model.VisionTransformer` and `timm.models.vision_transformer.VisionTransformer`. If your model comes from a different source, you will need to modify `class Peft_ViT`. At the very least, you should check and possibly rewrite the `__init__` function. You can refer to [lines 259-280](https://github.com/shijxcs/LIFT/blob/main/models/peft_vit.py#L259-L280) as examples.

**Comment by Lavi11C** [comment]
I don't understand what different source mean. Because I used vit-l14 as well, I thought that meant same source, didn't it?

**Comment by shijxcs** [comment]
Different sources mean that different code bases may have different implementations of vit-l14, for example, the variable names may be different. In this case, you will need to rewrite `class Peft_ViT` to identify certain layers (e.g. MHSA, FFN) to insert lightweight modules.

**Comment by Lavi11C** [comment]
Thanks to remind me this thing. If I don't identify layer certainly, I will get error code or low accuracy, right?

**Comment by shijxcs** [comment]
Yes, you are right. It will most likely throw an error if your `vit_model` is not an instance of `CLIP_ViT` or `ViT`, as neither of the conditions on [lines 259-280](https://github.com/shijxcs/LIFT/blob/main/models/peft_vit.py#L259-L280) will be satisfied. A special case is when the `vit_model` is a subclass of `CLIP_ViT` or `ViT`, which won't cause an error. However, in this case, you should check if other details need adjustment, otherwise it may lead to a low accuracy.

**Comment by Lavi11C** [comment]
Ok, I will check this thing. Thank you very much.
